functional_tests/test_db_online_offline_resync.py

```diff
@@ -0,0 +1,441 @@
+    recieved_docs = in_parallel(user_objects, 'get_num_docs')
+
+    expected_docs = num_users * num_docs
+    for user_obj, docs in recieved_docs.items():
+        log.info('User {} got {} docs, expected docs: {}'.format(user_obj.name, docs, expected_docs))
+        assert docs == 1 + expected_docs
+
+
+    # Verify that
+    # user created doc-ids exist in docs received in changes feed
+    # expected revision is equal to received revision
+    expected_revision = str(num_revisions + 1)
+    docs_rev_dict = in_parallel(user_objects, 'get_num_revisions')
+    rev_errors = []
+    for user_obj, docs_revision_dict in docs_rev_dict.items():
+        for doc_id in docs_revision_dict.keys():
+            rev = docs_revision_dict[doc_id]
+            log.info('User {} doc_id {} has {} revisions, expected revision: {}'.format(user_obj.name,
+                                                                                        doc_id, rev, expected_revision))
+            if rev != expected_revision:
+                rev_errors.append(doc_id)
+                log.error('User {} doc_id got revision {}, expected revision {}'.format(user_obj.name,
+                                                                                        doc_id, rev, expected_revision))
+
+    assert len(rev_errors) == 0
+
+    # Verify each User created docs are part of changes feed
+    output = in_parallel(user_objects, 'check_doc_ids_in_changes_feed')
+    assert True in output.values()
+
+    # Take "db" offline
+    status = admin.take_db_offline(db="db")
+    assert(status == 200)
+
+    restart_status = cluster.sync_gateways[0].restart("bucket_online_offline/db_online_offline_access_restricted.json")
+    assert restart_status == 0
+
+    time.sleep(10)
+
+    num_changes = admin.db_resync(db="db")
+    log.info("expecting num_changes {} == num_docs {} * num_users {}".format(num_changes, num_docs, num_users))
+    assert(num_changes['changes'] == num_docs * num_users)
+
+    status = admin.bring_db_online(db="db")
+    assert(status == 200)
+
+    verify_changes(user_x, expected_num_docs=expected_docs, expected_num_revisions=num_revisions, expected_docs=user_objects[0].cache)
+
+    end = time.time()
+    log.info("Test ended.")
+    log.info("Main test duration: {}".format(end - init_completed))
+    log.info("Test setup time: {}".format(init_completed - start))
+    log.info("Total Time taken: {}s".format(end - start))
+
+
+# implements scenario: 11
+# With DB in online state, put a large number of docs (enough to cause _resync to run for 10-15 seconds),
+# put DB offline, run _resync, attempt to bring DB online while _resync is running,
+# expected result _online will fail with status 503, when _resync is complete,
+# attempt to bring DB _online, expected result _online will succeed, return status 200.
+@pytest.mark.sanity
+@pytest.mark.dbonlineoffline
+@pytest.mark.parametrize("num_users", [5])
+@pytest.mark.parametrize("num_docs", [100])
```

Comment:
It was enough to run the resync and do the test operation within the time. Adding more docs will just run resync longer but will not have any impact on the test condition